enhancements/network/baremetal-networking.md

```diff
@@ -0,0 +1,260 @@
+reachable on the bootstrap instance at first and then later on the
+newly-deployed control plane machines.
+
+#### API VIP (Virtual IP)
+
+In cluster network infrastructure, a VIP (Virtual IP) is used to provide
+failover of the API server across the control plane machines
+(including the bootstrap instance). This "API VIP" is provided by the user
+as an `install-config.yaml` [parameter](https://github.com/openshift/installer/blob/master/pkg/types/baremetal/platform.go#L57-L63)
+and the installation process configures `keepalived` to manage this VIP.
+
+The API VIP first resides on the bootstrap instance.
+This `keepalived` [instance](https://github.com/openshift/machine-config-operator/blob/master/manifests/baremetal/keepalived.yaml)
+are run as a [static pod](https://kubernetes.io/docs/tasks/administer-cluster/static-pod/) and the
+[relevant assets](https://github.com/openshift/machine-config-operator/blob/master/manifests/baremetal/keepalived.conf.tmpl#L7)
+are rendered by the Machine Config Operator.
+
+The VIP will move to one of the control plane nodes, but only after the
+bootstrap process has completed and the bootstrap instance is stopped. This happens
+because the `keepalived` [instances](https://github.com/openshift/machine-config-operator/blob/master/templates/common/baremetal/files/baremetal-keepalived.yaml)
+on control plane machines are [configured](https://github.com/openshift/machine-config-operator/blob/master/templates/common/baremetal/files/baremetal-keepalived.yaml)
+with a lower [VRRP](https://en.wikipedia.org/wiki/Virtual_Router_Redundancy_Protocol)
+priority. This ensures that the API on the control plane nodes is fully
+functional before the API VIP moves.
+
+#### API load balancing
+
+Once the API VIP has moved to one of the control plane nodes, traffic sent from
+external clients to this VIP first hits an `haproxy` load balancer running on
+that control plane node.
+These [instances](https://github.com/openshift/machine-config-operator/blob/master/templates/master/00-master/baremetal/files/baremetal-haproxy.yaml)
+of `haproxy` are [configured](https://github.com/openshift/machine-config-operator/blob/master/templates/master/00-master/baremetal/files/baremetal-haproxy-haproxy.yaml)
+to load balance the API traffic across all of the control plane nodes.
+
+### Internal DNS
+
+Externally resolvable DNS records are required for:
+
+* `api.$cluster_name.$base-domain` -
+* `*.apps.$cluster_name.$base_domain` -
+
+In addition, internally resolvable DNS records are required for:
+
+* `api-int.$cluster_name.$base-domain` -
+* `etcd-NNN.$cluster_name.$base-domain` -
+* SRV record: `_etcd-server-ssl._tcp.$cluster_name.$base-domain` - Lists the
+  `etcd-NNN` records.
+
+In cluster networking infrastructure, the goal is is to automate as much of the
+DNS requirements internal to the cluster as possible, leaving only a
+small amount of public DNS configuration to be implemented by the user
+before starting the installation process.
```

Comment:
> leaving only a
> small amount of public DNS configuration to be implemented by the user
> before starting the installation process.

So the customer should be able to use the kubeconfig provided without any pre/post setup.

The only one that is kinda acceptable is the `*.apps.$cluster_domain` from outside the cluster.

Reply:
@abhinavdahiya I would think that the A record for api and *.app would need to be create before starting install.

Reply:
can to add this to limitations.